nessie/src/main/java/org/apache/iceberg/nessie/TableReference.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.nessie;
+
+import java.time.Instant;
+import org.apache.iceberg.catalog.TableIdentifier;
+
+public class TableReference {
+
+  private final TableIdentifier tableIdentifier;
+  private final Instant timestamp;
+  private final String reference;
+
+  /**
+   * Container class to specify a TableIdentifier on a specific Reference or at an Instant in time.
+   */
+  public TableReference(TableIdentifier tableIdentifier, Instant timestamp, String reference) {
+    this.tableIdentifier = tableIdentifier;
+    this.timestamp = timestamp;
+    this.reference = reference;
+  }
+
+  public TableIdentifier getTableIdentifier() {
```

Comment:
Nit: We usually omit `get` from getter method names because it doesn't add value and looks weird in other languages (like Scala and Kotlin).

Reply:
fixed. The pythonista in me doesn't like `get` either :-)